### ISSUE TITLE
PDF transform: scale output exception

### ISSUE THREAD

**Issue by sqcddzx** [puppeteer/puppeteer#5162]
### Steps to reproduce

**Tell us about your environment:**

* Puppeteer version: v2.0.0
* Platform / OS version: win10
* Node.js version: v10.14.2

**What steps will reproduce the problem?**
    First, I setup a basic test page like this:

    let html = `<!DOCTYPE html>
    <html lang="en">
    <head>
      <meta charset="UTF-8">
      <title>Document</title>
      <style>
        body{
          padding: 0;
          margin: 0;
        }

        #test{
          width: 200px;
          font-size: 24px;
          transform: scale(0.8);
        }
      </style>
    </head>
    <body>
      <div id="test">测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测测测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测</div>
    </body>
    </html>`

    const puppeteer = require('puppeteer')

    ;(async () => {
      let browser = await puppeteer.launch({
        devtools: false
      })

      let page = await browser.newPage()

      await page.setContent(html)

      await page.pdf({
        path: './test.pdf', 
        width: 1000,
        height: 1010 
      })
    })()

**What is the expected result?**

The PDF output is consistent with **devtools** model

**What happens instead?**
First
  In the '#test' transform: scale (1);
  **When the height of a single text box exceeds the set page height.**
  The excess section will be divided into the second page with spacing between the two pages (**less than the font height**).
![image](https://user-images.githubusercontent.com/6456329/68825117-f9fa0680-06d3-11ea-9f76-aa78cfbd6c3e.png)

And Then
  '#test' transform: scale (0.8)  
  **When you scale it, it's still there**.

![image](https://user-images.githubusercontent.com/6456329/68825289-92908680-06d4-11ea-857a-59d270042423.png)

The devtools pattern looks like this, and I want the PDF output to look like this
![image](https://user-images.githubusercontent.com/6456329/68825871-b785f900-06d6-11ea-865b-0934247faec2.png)

**Comment by kreeble** [comment]
Very similar to a problem I experienced with PDF output having random, incorrect vertical gaps.
Resolved by avoiding CSS transform and using some other positioning workaround instead.

Wish I had found your Github issue before searching many other issues haha! This issue title is accurate, but nearly impossible to find without already having spent a long time ruling out many other possibilities and getting pretty close to the actual source of the issue.
